### PR TITLE
add ruff dependency and upgrade requirements

### PR DESCRIPTION
**Pull request by Mitchina**

### CODE REVIEW

**Comment by trottomv** [comment]
the `Makefile` changes are missing, those should be added as well

File: {{cookiecutter.project_dirname}}/pyproject.toml [hunk 1]
```diff
@@ -21,28 +21,55 @@ omit = [
...
 DJANGO_CONFIGURATION = "Testing"
 DJANGO_SETTINGS_MODULE = "{{ cookiecutter.django_settings_dirname }}.settings"
+
+[tool.ruff]
+extend-exclude = [
+    "__pycache__",
+    ".pytest_cache",
+    ".vscode*",
+    "*/migrations/*",
+    "*/tests/data/*",
```
Comment (pauloxnet) [inline {{cookiecutter.project_dirname}}/pyproject.toml:42]:
```suggestion
```
Remove this project specific directory

File: {{cookiecutter.project_dirname}}/requirements/base.in [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # pip-compile --output-file base.txt base.in
-django-configurations[cache,database,email]~=2.3.0
-django~=4.0.0
+django-configurations[cache,database,email]~=2.4 .0
```
Comment (pauloxnet) [inline {{cookiecutter.project_dirname}}/requirements/base.in:2]:
```suggestion
django-configurations[cache,database,email]~=2.4.0
```
Remove wrong space

File: {{cookiecutter.project_dirname}}/requirements/remote.in [hunk 3]
```diff
@@ -1,9 +1,10 @@
...
 -r common.in
-{% if "s3" in cookiecutter.media_storage %}django-storages[boto3]~=1.12.0
+argon2-cffi~=21.3.0
+{% if "s3" in cookiecutter.media_storage %}django-storages[boto3]~=1.13.0
 {% endif %}gunicorn~=20.1.0
-{% if cookiecutter.use_redis == "true" %}redis~=4.1.0
-{% endif %}sentry-sdk~=1.5.0
-uvicorn[standard]~=0.17.0
+{% if cookiecutter.use_redis == "true" %}redis~=4.4.0
+{% endif %}sentry-sdk~=1.11.0
```
Comment (pauloxnet) [inline {{cookiecutter.project_dirname}}/requirements/remote.in:7]:
```suggestion
{% endif %}sentry-sdk~=1.12.0
```

File: {{cookiecutter.project_dirname}}/requirements/remote.in [hunk 4]
```diff
@@ -1,9 +1,10 @@
...
+uvicorn[standard]~=0.20.0
 wait-for-it~=2.2.0
-whitenoise[brotli]~=5.3.0
+whitenoise[brotli]~=6.2.0
```
Comment (pauloxnet) [inline {{cookiecutter.project_dirname}}/requirements/remote.in:10]:
```suggestion
whitenoise[brotli]~=6.3.0
```

**Review by pauloxnet — CHANGES_REQUESTED** [review]

File: {{cookiecutter.project_dirname}}/.gitignore [hunk 5]
```diff
@@ -168,15 +168,22 @@ cython_debug/
...
 
+# Ruff
+.ruff_cache
+
 # Others
-*.bak
 .bash_history
-.local
 .ipython
+.kolo
```
Comment (pauloxnet) [inline {{cookiecutter.project_dirname}}/.gitignore:177]: Move this row in #258

File: {{cookiecutter.project_dirname}}/.gitignore [hunk 6]
```diff
@@ -168,15 +168,22 @@ cython_debug/
...
+.local
+.tfscan
+.trivy
```
Comment (pauloxnet) [inline {{cookiecutter.project_dirname}}/.gitignore:180]:
```suggestion
```
Move this two rows in https://github.com/20tab/django-continuous-delivery/pull/250

File: {{cookiecutter.project_dirname}}/requirements/local.in [hunk 7]
```diff
@@ -1,14 +1,12 @@
...
-django-extensions~=3.1.0
-django-upgrade~=1.5.0
+django-debug-toolbar~=3.8.0
+django-extensions~=3.2.0
 graphviz~=0.20.0
-ipdb~=0.13.0
-pip-tools~=6.6.0
-pre-commit~=2.18.0
+ipython~=8.8.0
+kolo~=2.3.0
```
Comment (pauloxnet) [inline {{cookiecutter.project_dirname}}/requirements/local.in:7]:
```suggestion
```
Move this row in #258

File: {{cookiecutter.project_dirname}}/requirements/remote.in [hunk 8]
```diff
@@ -1,9 +1,10 @@
 # pip-compile --output-file remote.txt remote.in
 -r common.in
-{% if "s3" in cookiecutter.media_storage %}django-storages[boto3]~=1.12.0
+argon2-cffi~=21.3.0
```
Comment (pauloxnet) [inline {{cookiecutter.project_dirname}}/requirements/remote.in:3]:
```suggestion
```
Move this row in #258

**Review by pauloxnet — CHANGES_REQUESTED** [review]

File: {{cookiecutter.project_dirname}}/requirements/test.in [hunk 9]
```diff
@@ -2,19 +2,15 @@
...
-coverage[toml]~=6.3.0
-flake8-assertive~=2.1.0
-flake8-bugbear~=22.4.25
-flake8-docstrings~=1.6.0
-flake8-isort~=4.1.0
-flake8~=4.0.0
-isort~=5.10.0
-mypy~=0.931
+black~=22.12.0
+checkov~=2.2.0
```
Comment (pauloxnet) [inline {{cookiecutter.project_dirname}}/requirements/test.in:6]:
```suggestion
```
Change already added in #251

File: {{cookiecutter.project_dirname}}/scripts/check.sh [hunk 10]
```diff
@@ -2,9 +2,10 @@
...
+python3 -m manage check
+python3 -m manage makemigrations --dry-run --check
 python3 -m black --check .
-python3 -m isort --check .
-python3 -m flake8
-python3 -m mypy .
+ruff .
 python3 -m bandit --quiet --recursive --exclude tests .
 python3 -m pip_audit --require-hashes --requirement requirements/remote.txt
+checkov --download-external-modules=false --evaluate-variables=true --skip-framework terraform --quiet -d .
```
Comment (pauloxnet) [inline {{cookiecutter.project_dirname}}/scripts/check.sh:11]:
```suggestion
```
Change already added in #251

**Review by pauloxnet — CHANGES_REQUESTED** [review]

**Comment by Mitchina** [comment]
@trottomv the makefile changes, if it was only the update command, is already present inside the makefile of the main branch, let me know if you were thinking about another command

File: {{cookiecutter.project_dirname}}/pyproject.toml (hunk 1 above)
Comment (Mitchina) [reply {{cookiecutter.project_dirname}}/pyproject.toml:42]: okk

File: {{cookiecutter.project_dirname}}/requirements/base.in (hunk 2 above)
Comment (Mitchina) [reply {{cookiecutter.project_dirname}}/requirements/base.in:2]: giusto

File: {{cookiecutter.project_dirname}}/requirements/remote.in (hunk 3 above)
Comment (Mitchina) [reply {{cookiecutter.project_dirname}}/requirements/remote.in:7]: okk

File: {{cookiecutter.project_dirname}}/requirements/remote.in (hunk 4 above)
Comment (Mitchina) [reply {{cookiecutter.project_dirname}}/requirements/remote.in:10]: okk

File: {{cookiecutter.project_dirname}}/.gitignore (hunk 5 above)
Comment (Mitchina) [reply {{cookiecutter.project_dirname}}/.gitignore:177]: okk

**Comment by trottomv** [comment]
> @trottomv the makefile changes, if it was only the update command, is already present inside the makefile of the main branch, let me know if you were thinking about another command

iIn the Makefile the `make check` and `make fix` commands should be modified with invocations of ruff instead of flake8 and isort

File: {{cookiecutter.project_dirname}}/.gitignore (hunk 6 above)
Comment (Mitchina) [reply {{cookiecutter.project_dirname}}/.gitignore:180]: okk

File: {{cookiecutter.project_dirname}}/requirements/local.in (hunk 7 above)
Comment (Mitchina) [reply {{cookiecutter.project_dirname}}/requirements/local.in:7]: okk

File: {{cookiecutter.project_dirname}}/requirements/remote.in (hunk 8 above)
Comment (Mitchina) [reply {{cookiecutter.project_dirname}}/requirements/remote.in:3]: okk

File: {{cookiecutter.project_dirname}}/requirements/test.in (hunk 9 above)
Comment (Mitchina) [reply {{cookiecutter.project_dirname}}/requirements/test.in:6]: okk, removed

**Review by niccolomineo — APPROVED** [review]

**Review by pauloxnet — CHANGES_REQUESTED** [review]

**Review by pauloxnet — DISMISSED** [review]

**Review by trottomv — APPROVED** [review]

File: {{cookiecutter.project_dirname}}/scripts/check.sh [hunk 11]
```diff
@@ -2,9 +2,9 @@
 
 set -euo pipefail
 
+python3 -m manage check
+python3 -m manage makemigrations --dry-run --check
 python3 -m black --check .
-python3 -m isort --check .
-python3 -m flake8
-python3 -m mypy .
+ruff .
```
Comment (niccolomineo) [inline {{cookiecutter.project_dirname}}/scripts/check.sh:8]:
```suggestion
python3 -m ruff .
```

File: {{cookiecutter.project_dirname}}/Makefile [hunk 12]
```diff
@@ -31,9 +31,8 @@ dumpgroups:  ## Django dump auth.Group data
 .PHONY: fix
 fix:  ## Fix code formatting, linting and sorting imports
 	python3 -m black .
-	python3 -m isort .
-	python3 -m flake8
-	python3 -m mypy .
+	ruff --fix .
```
Comment (niccolomineo) [inline {{cookiecutter.project_dirname}}/Makefile:34]:
```suggestion
	python3 -m ruff --fix .
```

File: {{cookiecutter.project_dirname}}/scripts/check.sh (hunk 11 above)
Comment (Mitchina) [reply {{cookiecutter.project_dirname}}/scripts/check.sh]: as agreed, I'm reverting it

File: {{cookiecutter.project_dirname}}/Makefile (hunk 12 above)
Comment (Mitchina) [reply {{cookiecutter.project_dirname}}/Makefile]: as agreed, I'm reverting it

**Review by pauloxnet — APPROVED** [review]

File: {{cookiecutter.project_dirname}}/Dockerfile [hunk 13]
```diff
@@ -1,16 +1,18 @@
-ARG PYTHON_VERSION=3.10
-FROM python:${PYTHON_VERSION}-slim-bullseye AS base
+FROM python:3.11-slim-bullseye AS base
```
Comment (filippo-20tab) [inline {{cookiecutter.project_dirname}}/Dockerfile:1]: Let's keep track of the reasons that led to the removal of the parametric Python version. This feature may be reintroduced at a later stage, after having deepened all the possible issues.

**Pull request merged**